core/scheduler/service/job_run_service.go

```diff
@@ -436,7 +437,11 @@ func (s *JobRunService) updateOperatorRun(ctx context.Context, event *scheduler.
 
 func (s *JobRunService) trackEvent(event *scheduler.Event) {
 	if event.Type.IsOfType(scheduler.EventCategorySLAMiss) {
-		s.l.Info("received job sla_miss event, jobName: %v , slaPayload: %#v", event.JobName, event.SLAObjectList)
+		jsonSLAObjectList, err := json.Marshal(event.SLAObjectList)
+		if err != nil {
+			jsonSLAObjectList = []byte("unable to json Marshal SLAObjectList")
```

Comment:
log & return immediately and please make all else if as, just if's and return

Reply:
actually along with logging we are also tracking these events on the Prometheus , hence we can not return early

Reply:
converted to switch case